src/js/components/Diagram/stories/Animated.js

```diff
@@ -66,58 +66,53 @@ const Container = ({ node, index }) => (
   />
 );
 
-class Animated extends React.Component {
-  state = {
-    draw: true,
-  };
+const Animated = () => {
+  const reducer = draw => !draw;
 
-  componentDidMount() {
-    this.timer = setInterval(() => {
-      const { draw } = this.state;
-      this.setState({
-        draw: !draw,
-      });
-    }, 3000);
-  }
+  const [draw, toogleDraw] = useReducer(reducer, true);
```

Comment:
I think that using a `useReducer` as an alternative to `useState` makes sense for a more complex state logic.
Hence I'm just wondering why you chose it vs `useState`, keeping in mind that the storybook components supposed to provide simple examples. let me know your thoughts.

Reply:
Imagine the following scenario:
```js
useEffect(() => {
    const id = setInterval(() => {
      setDraw(!draw);
    }, 3000);
    return () => clearInterval(id);
  }, []);
```

This should create the setInterval at the start right? But if you look closely you arent passing the `draw` dependency to useEffect, you are kinda cheating react, this will create an unexpected behaviour.
It will work like this: 
```
setValue(!true) every 2seconds

first State -> draw === true
2s after -> draw === false
2s after -> draw === false
```

The component isnt re-rerendering, the useEffect its always use the `draw === true` state to change the state every interval tick.

The solution would be:
```js
useEffect(() => {
    const id = setInterval(() => {
      setDraw(!draw);
    }, 3000);
    return () => clearInterval(id);
  }, [draw]);
```
Right??
However it wouldnt work because you are triggering the effect to change `draw` when `draw` is changed. See the problem?
By using `useReducer`, at render it calls the dispatcher to alters its state. This way you keep the dependencies on useEffect clear.

with useReducer it work like this:
```
Every 3 seconds the toogleDraw function will be called and it will invert the current `draw` state
```

Reply:
Thank you for the detailed explanation, nice work